src/Domain/templates/emails/customer-new-account-blocks.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Customer new account email - html.
+ *
+ * This is intended as a replacement to WC_Email_Customer_New_Account(),
+ * with a set password link instead of including the new password in email
+ * content.
+ *
+ * @package  WooCommerce/Blocks
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+do_action( 'woocommerce_email_header', $email_heading, $email ); ?>
+
+<?php /* translators: %s: Customer username */ ?>
+<p><?php printf( esc_html__( 'Hello %s,', 'woo-gutenberg-products-block' ), esc_html( $user_login ) ); ?></p>
+<?php /* translators: %1$s: Site title, %2$s: Username, %3$s: My account link */ ?>
+<p><?php printf( esc_html__( 'Thanks for creating an account on %1$s. Your username is %2$s. You can access your account area to view orders, change your password, and more at: %3$s', 'woo-gutenberg-products-block' ), esc_html( $blogname ), '<strong>' . esc_html( $user_login ) . '</strong>', make_clickable( esc_url( wc_get_page_permalink( 'myaccount' ) ) ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?></p>
+<?php if ( $set_password_url ) : ?>
+	<p><a href="<?php echo esc_attr( $set_password_url ); ?>"><?php printf( esc_html__( 'Click here to set your new password.', 'woo-gutenberg-products-block' ) ); ?></a></p>
+<?php endif; ?>
```

Comment:
I realize this is just the first iteration, but I think we'll want to improve on the wording a bit. It's entirely possible the user won't be able to access their account if the login session has expired. So we probably shouldn't direct them to the account page in this new initial email but instead just direct them to set their new password (same applies to the plain email template).

Reply:
Agree - this really needs design. I'm currently using a hybrid of the existing woo new account templates and the wording from core for the set initial password section (i.e. combo of all prior art).

Note that the generic "my account" link should be useful since some users may use the email as a way to get back to their account; the reset password link will expire, but my account will always show login form, reset password options so user can get back in. I'm guessing this is part of the reason it's included in these emails.